### PR TITLE
Updates for global registry support

### PR DESCRIPTION
**Pull request by gitlawr**
1. Add `global-registry-enabled` setting, It is used to detect whether global registry is enabled for non-admin users.
2. Copy ca cert to `/etc/docker/certs.d/<server-host>[:port]/ca.crt` on the host([see reference](https://docs.docker.com/engine/security/certificates/)). So that nodes on clusters can connect to the global registry in custom CA root cases. CA cert locates in `/etc/kubernetes/ssl/certs/serverca` both inside node agents & on the hosts before, this keeps the same.
  In this solution, `/etc/docker/certs.d` instead of `/etc/docker/certs.d/<server-host>[:port]` on the host mounts node agent containers, due to two limitations:
  - Colon is not supported in kuberentes volume path, which may be needed in docker cert path. [see](https://github.com/kubernetes/kubernetes/issues/21681#issuecomment-187260705)
  - hostpath-subpath is broken, And looks like we need a deprecated kubelet flag for [that](https://github.com/rancher/rancher/issues/14836).

Depending on the system-chart PR: https://github.com/rancher/system-charts/pull/41

https://github.com/rancher/rancher/issues/20597

### CODE REVIEW

**Comment by orangedeng** [comment]
I think we also need a registry template id in settings also.

**Comment by gitlawr** [comment]
Updated

File: package/run.sh [hunk 1]
```diff
@@ -244,8 +245,8 @@ if [ -n "$CATTLE_CA_CHECKSUM" ]; then
         error "Please check if the correct certificate is configured at $CATTLE_SERVER/v3/settings/cacerts"
         exit 1
     else
-        mkdir -p /etc/kubernetes/ssl/certs
-        mv $temp /etc/kubernetes/ssl/certs/serverca
+        mkdir -p /etc/kubernetes/ssl/certs/$CATTLE_SERVER_HOSTNAME_WITH_PORT
+        mv $temp /etc/kubernetes/ssl/certs/$CATTLE_SERVER_HOSTNAME_WITH_PORT/ca.crt
```
Comment (orangedeng) [inline package/run.sh]: I think subdirectory in `/etc/kubernetes/ssl/certs/` won't work. In the default cert pool of golang https://github.com/golang/go/blob/29907b13db0455eded50263b4e37445045c82e6e/src/crypto/x509/root_unix.go#L64-L82, only the cert files in `/etc/kubernetes/ssl/certs/` can be loaded but not those in subdir.

Comment (gitlawr) [reply package/run.sh]: Updated. Now we are not touching the `serverca` file, but copying cacert to `/etc/docker/certs.d` additionally

**Comment by orangedeng** [comment]
LGTM

**Comment by loganhz** [comment]
Link to https://github.com/rancher/rancher/issues/1738

**Comment by gitlawr** [comment]
Removed the template id setting per discussion.

**Comment by loganhz** [comment]
LGTM

File: pkg/settings/setting.go [hunk 2]
```diff
@@ -24,6 +24,7 @@ var (
 	EngineNewestVersion             = NewSetting("engine-newest-version", "v17.12.0")
 	EngineSupportedRange            = NewSetting("engine-supported-range", "~v1.11.2 || ~v1.12.0 || ~v1.13.0 || ~v17.03.0 || ~v17.06.0 || ~v17.09.0 || ~v18.06.0 || ~v18.09.0")
 	FirstLogin                      = NewSetting("first-login", "true")
+	GlobalRegistryEnabled           = NewSetting("global-registry-enabled", "false")
```
Comment (cjellick) [inline pkg/settings/setting.go]: Who sets this setting to true?
How is it used

Comment (loganhz) [reply pkg/settings/setting.go]: UI will use it. UI need to know if global registry is enabled to decide if we should global registry is an option in adding registry credential page

File: tests/validation/tests/v3_api/test_global_registry.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
+import pytest
```
Comment (cjellick) [inline tests/validation/tests/v3_api/test_global_registry.py]: Moving forward, lets do validation tests as separate PRs so that they can be easily flagged for QA team to review

Comment (loganhz) [reply tests/validation/tests/v3_api/test_global_registry.py]: Got it

**Comment by cjellick** [comment]
Backend code is LGTM, but I am worried about the test a little bit. I'd like you to remove it from this PR and open up in its own PR. I want @sangeethah and @bmdepesa to review it for potential impact to normal test runs.

**Comment by gitlawr** [comment]
Moved the test commit to #20605

**Comment by cjellick** [comment]
lgtm
